File: Python/7py 1/generate_shape.py
"""
|--------------------------------------------------------------------------
| Build a square
|--------------------------------------------------------------------------
|
"""

def generate_shape(n):
   res = []
   for i in range(n):
      res.append('+' * n)
   return '\n'.join(res)


def generateShape(integer):
   return '\n'.join('+' * integer for i in range(integer))


def generateShape(i):
   return (i-1)*(('+'*i)+'\n')+('+'*i)


def generateShape(integer):
   return (f"{'+' * integer}\n" * integer)[:-1]


print(generate_shape(3))
print(generate_shape(8))

File: PHP/generateShape.php
<?php
/*
|--------------------------------------------------------------------------
| Build a square
|--------------------------------------------------------------------------
|
*/

function generateShape($int) {
   $res = "";
   for ($i=0; $i < $int; $i++) { 
      for ($j=0; $j < $int; $j++) { 
         $res .= '+';
      }
      if ($i != $int - 1) {
         $res .= "\n";
      }
   }
   return $res;
}


function generateShape($int) {
   return trim(str_repeat(str_repeat('+', $int) . "\n", $int));
}


function generateShape($int) {
   return substr(str_repeat(str_repeat("+", $int) . "\n", $int), 0, strlen(str_repeat(str_repeat("+", $int) . "\n", $int)) - 1);
}


function generateShape($n): string {
   return implode("\n", array_fill(0, $n, str_repeat('+', $n)));
}


print_r(generateShape(8));
print_r(generateShape(3));